File: src/main/java/com/uitnetwork/alexa/util/AppUtils.java
package com.uitnetwork.alexa.util;

import static com.uitnetwork.alexa.util.AppConstants.ENV_VARIABLE_SUPPORTED_APP_IDS;
import static com.uitnetwork.alexa.util.AppConstants.SUPPORTED_APP_ID_SEPERATOR;
import static java.lang.System.getenv;
import static java.util.Arrays.asList;
import static org.slf4j.LoggerFactory.getLogger;

import java.util.HashSet;
import java.util.Set;

import org.slf4j.Logger;

import com.uitnetwork.alexa.service.TimeIntentService;

public class AppUtils {
    private static final Logger logger = getLogger(TimeIntentService.class);


    public static final Set<String> getSupportedAppIds() {
        String supportedAppIdsString = getenv(ENV_VARIABLE_SUPPORTED_APP_IDS);

        String[] supportedAppIdArray = supportedAppIdsString.split(SUPPORTED_APP_ID_SEPERATOR);
        HashSet<String> supportedAppIds = new HashSet<>(asList(supportedAppIdArray));

        logger.info("Supported App Ids are: {}", supportedAppIdsString);
        return supportedAppIds;
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/util/AppUtils.kt
package com.uitnetwork.alexa.util

import com.uitnetwork.alexa.service.TimeIntentService
import com.uitnetwork.alexa.util.AppConstants.Companion.ENV_VARIABLE_SUPPORTED_APP_IDS
import com.uitnetwork.alexa.util.AppConstants.Companion.SUPPORTED_APP_ID_SEPERATOR
import org.slf4j.LoggerFactory.getLogger
import java.lang.System.getenv
import java.util.*
import java.util.Arrays.asList

object AppUtils {
    private val logger = getLogger(TimeIntentService::class.java)


    val supportedAppIds: Set<String>
        get() {
            val supportedAppIdsString = getenv(ENV_VARIABLE_SUPPORTED_APP_IDS)

            val supportedAppIdArray = supportedAppIdsString.split(SUPPORTED_APP_ID_SEPERATOR.toRegex()).dropLastWhile({ it.isEmpty() }).toTypedArray()
            val supportedAppIds = HashSet<String>(asList<String>(*supportedAppIdArray))

            logger.info("Supported App Ids are: {}", supportedAppIdsString)
            return supportedAppIds
        }
}
